packages/strapi-plugin-content-manager/controllers/collection-types.js

```diff
@@ -225,4 +226,57 @@ module.exports = {
 
     ctx.body = results.map(result => permissionChecker.sanitizeOutput(result));
   },
+
+  async previewManyRelations(ctx) {
+    const { userAbility } = ctx.state;
+    const { model, id, targetField } = ctx.params;
+    const { pageSize = 10, page = 1 } = ctx.request.query;
+
+    const contentTypeService = getService('content-types');
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.read()) {
+      return ctx.forbidden();
+    }
+
+    const modelDef = strapi.getModel(model);
+    const assoc = modelDef.associations.find(a => a.alias === targetField);
+
+    if (!assoc || !MANY_RELATIONS.includes(assoc.nature)) {
+      return ctx.badRequest('Invalid target field');
+    }
+
+    const entity = await entityManager.findOneWithCreatorRoles(id, model);
+
+    if (!entity) {
+      return ctx.notFound();
+    }
+
+    if (permissionChecker.cannot.read(entity, targetField)) {
+      return ctx.forbidden();
+    }
+
+    let relationList;
+    if (assoc.nature === 'manyWay') {
+      const populatedEntity = await entityManager.findOne(id, model, [targetField]);
+      const relationsListIds = populatedEntity[targetField].map(prop('id'));
+      relationList = await entityManager.findPage(
+        { page, pageSize, id_in: relationsListIds },
```

Comment:
@alexandrebodin We added the `page` and `pageSize` in the restricted attributes because of this queries. What do you think?

Reply:
the page / pageSize parameters are interpreted before querying the db (in strapi-database) so I don't see why we would need to restrict them ? if the fields exist the filtering will be done with _where: { page: x, pageSize: y  } for example so it is compatible.